js/util/segment-fetcher.js

```diff
@@ -20,117 +20,14 @@
- * - `SEGMENT_VERIFICATION_FAILED`: if any retrieved segment fails
- *   the user-provided VerifySegment callback or KeyChain verifyData.
- * - `IO_ERROR`: for I/O errors when sending an Interest.
- *
- * In order to validate individual segments, a KeyChain needs to be supplied.
- * If verifyData fails, the fetching process is aborted with
- * SEGMENT_VERIFICATION_FAILED. If data validation is not required, pass null.
- *
- * Example:
- *     var onComplete = function(content) { ... }
- *
- *     var onError = function(errorCode, message) { ... }
- *
- *     var interest = new Interest(new Name("/data/prefix"));
- *     interest.setInterestLifetimeMilliseconds(1000);
- *
- *     SegmentFetcher.fetch(face, interest, null, onComplete, onError);
- *
- * This is a private constructor to create a new SegmentFetcher to use the Face.
- * An application should use SegmentFetcher.fetch. If validatorKeyChain is not
- * null, use it and ignore verifySegment. After creating the SegmentFetcher,
- * call fetchFirstSegment.
- * @param {Face} face This calls face.expressInterest to fetch more segments.
- * @param validatorKeyChain {KeyChain} If this is not null, use its verifyData
- * instead of the verifySegment callback.
- * @param {function} verifySegment When a Data packet is received this calls
- * verifySegment(data) where data is a Data object. If it returns False then
- * abort fetching and call onError with
- * SegmentFetcher.ErrorCode.SEGMENT_VERIFICATION_FAILED.
- * NOTE: The library will log any exceptions thrown by this callback, but for
- * better error handling the callback should catch and properly handle any
- * exceptions.
- * @param {function} onComplete When all segments are received, call
- * onComplete(content) where content is a Blob which has the concatenation of
- * the content of all the segments.
- * NOTE: The library will log any exceptions thrown by this callback, but for
- * better error handling the callback should catch and properly handle any
- * exceptions.
- * @param {function} onError Call onError.onError(errorCode, message) for
- * timeout or an error processing segments. errorCode is a value from
- * SegmentFetcher.ErrorCode and message is a related string.
- * NOTE: The library will log any exceptions thrown by this callback, but for
- * better error handling the callback should catch and properly handle any
- * exceptions.
- * @constructor
- */
-var SegmentFetcher = function SegmentFetcher
-  (face, validatorKeyChain, verifySegment, onComplete, onError)
-{
-  this.face = face;
-  this.validatorKeyChain = validatorKeyChain;
-  this.verifySegment = verifySegment;
-  this.onComplete = onComplete;
-  this.onError = onError;
-
-  this.contentParts = []; // of Buffer
-};
+var SegmentFetcher = function SegmentFetcher() { };
 
 exports.SegmentFetcher = SegmentFetcher;
 
-/**
- * An ErrorCode value is passed in the onError callback.
- */
-SegmentFetcher.ErrorCode = {
```

Comment:
SegmentFetcher.ErrorCode is public API (referenced by SegmentFetcher.fetch jsdoc). It must be preserved.